### ISSUE TITLE
POC  or Functionality Review, feasibility analysis on trial versions of PDFTron , PSPDFKIT

### ISSUE THREAD

**Issue by abin-aot** [bcgov/foi-flow#2791]
Title of ticket: 

#### Description
The purpose of this task is to review PDFTron's and PSPDFKIT React based trial version to tally Redaction, Harms assessment, Red Lining, Highlighting, Annotations/Comments, Search , Performance loading/LOD etc. Consider a fairly large PDF document , that has 1000(?) pages and create a sample 'create-react-app' type of web application, then load document from S3 using PDFTron . After that complete below mentioned DODs.

Authentication is not in scope. But better to understand , if we can control pages and viewing using code by auth. 

#### Dependencies
Are there any dependencies?

#### DOD
- [x]  Create a sample web app using 'create-react-app' and install PDFTRon/PSPDFKit Trial version. Contact 
- [x]  Check the PDF page loading time using PDFTron. Analyze the load time, understand whether there are any LOAD on Demand options, with in the tool
- [x]  Red Lining support - Saving, removing redline options. How redlining is saved ? With document or as extra metadata. 
- [x]  Highlighting support - Can we **_highlight multiple sections on each page_**   _that span across 1000 pages_ on our  
  large document . Means once AI/NLP is in place, we will be getting a array NER outputs on PII or any other keywords which need to be highlighted
- [x]  Redaction support - Implement PDFTRON based redaction support on to the document. **Try to save the document, and then download it to parse that using any external tool to see whether data has been completely** removing within PDF Layers. **_Can we Redact highlighted and/or redlined section in a single go ?_**
- [x] Comments, Annotations on Sections of document . How is save comments, Annotations work with respect to our BE design? Means, is that part of whole document or can be separated out from document as a metadata per page.
- [x]  Searching a text /Keyword on the document. Can we drive this from outside , like passing via code, array ?
- [x] Other PDF editing capabilities - means adding new pages, sections etc. Needed??
- [x] In the end, idea is to **_group the necessary features and prepare for review for licensing and cost involved_**.
- [x] Compare features on PDFTron vs PSPDFKIT - Its usability , customization , performance etc. 
- [x] Deploy it somewhere or Local  ? @lmullane  ?
- [x] Demo all DODs and comparison it to entire team, @lmullane @JHarrietha-AOT , @m-prodan

**Comment by abin-aot** [comment]
POC demo'd to entire team with code , UX and functionality. Compared both tools with required functionalities as indicated on the  DODs. Comparison chart shared with @lmullane , @m-prodan .

POC Code : https://bcgov.sharepoint.com/:u:/t/03521/EXaUnW009YVFrPVl1ogRQz8BiHvbZDZMuOCzwOIi4d8bog?e=LwekyV

Comparison Doc : https://bcgov.sharepoint.com/:x:/t/03521/EYq3sKR-WtlFsjSVrFRzNH4BDU0ak9uqFSKgJlGOZYQXOA?e=n3KnBU
